ui/ducks/send/send.js

```diff
@@ -1937,6 +1951,38 @@ export function updateRecipient({ address, nickname }) {
   };
 }
 
+/**
+ * This method is called when user changes the network in the send page.
+ * It doesn't update the chain id for transaction, rather it reject the existing and create new transaction instead
+ * then it assigns the new transaction id to the draft transaction id.
+ */
+export function onNetworkChange() {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const draftTxId = state[name].currentTransactionUUID;
+    const existingTransactionId = state[name].draftTransactions[draftTxId]?.id;
+
+    if (!existingTransactionId) {
+      // This means there is no transaction to reject.
+      return;
+    }
+
+    const existingTransaction = state.metamask.transactions.find(
```

Comment:
New chain id will be in the state of `NetworkController` only if this `onNetworkChange` call is right after dispatch of awaiting network change. So I was assuming this is guaranteed. Since `addTransaction` gets chain id by calling `NetworkController:getNetworkClientById`, I think it's safe if it's in the right order. If not we won't be able to get current chain id here anyway.

This is definitely doable but wouldn't this be safer to mimic what we are doing in the ultimate action in the send page which is using `addTransaction`?
My worry here is network change is shouldn't be something we update in a transaction because of missing certain validations (e.g https://github.com/MetaMask/core/blob/main/packages/transaction-controller/src/TransactionController.ts#L705) or doing `securityProviderRequest` etc.

Reply:
Sorry for the confusion, I'm not suggesting we update the chain ID on an existing transaction.

But you're absolutely right, the chain ID doesn't need to be set as set using the network state.

Reply:
Ah sorry my misunderstanding